s3store/s3.go

```diff
@@ -0,0 +1,47 @@
+package s3store
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// Store uses s3 as backing store
+type Store struct {
+	Bucket string
+}
+
+// Upload fulfills attache.Store interface
+func (s Store) Upload(file *bytes.Reader, fileType string) (string, error) {
+	fileName := filename(fileType)
+	filePath := fmt.Sprintf("https://s3-%s.amazonaws.com/%s/%s", os.Getenv("AWS_REGION"), s.Bucket, fileName)
+
+	// unsure about how long we can cache `svc` or must we really
+	// session.New everytime?
+	svc := s3.New(session.New())
+	_, err := svc.PutObject(&s3.PutObjectInput{
+		Bucket: aws.String(s.Bucket),
+		Body:   file,
+		Key:    &fileName,
+	})
+
+	return filePath, err
+}
+
+func filename(fileType string) string {
+	// Sorts in Reverse Chrono Order
+	key := strconv.FormatInt((math.MaxInt64 - time.Now().UnixNano()), 10)
```

Comment:
I know it's already using `UnixNano`... but would feel more sure footed, using some sort of UUID e.g. https://github.com/oklog/ulid

Reply:
in fact, if we can suffix with the sanitized version of the uploaded filename, it would also help operationally to eyeball and see which image is which. utterly optional though

Reply:
Am changing it to use `ulid` now.

Reply:
Am thinking that this method should go out to `store.go`?

Reply:
👍 while implementing `gcloudstore.Store` I did feel this can be a common function for various storage engines to use. For edge cases where this cannot work, then storage can 
write its own

> Am thinking that this method should go out to store.go?

You pick any file, but it should be a standalone, public function e.g. `attache.UniqueFilename()` not attached to the `attache.Store` interface.

Reply:
👍  New PR.